daemon/containerd/image_exporter.go

```diff
@@ -148,6 +149,12 @@ func leaseContent(ctx context.Context, store content.Store, leasesManager leases
 // complement of ExportImage.  The input stream is an uncompressed tar
 // ball containing images and metadata.
 func (i *ImageService) LoadImage(ctx context.Context, inTar io.ReadCloser, outStream io.Writer, quiet bool) error {
+	decompressed, err := dockerarchive.DecompressStream(inTar)
```

Comment:
Any reason not to use https://pkg.go.dev/github.com/containerd/containerd@v1.7.7/archive/compression#DecompressStream ?

Reply:
The only reason I can see is that this implementation doesn't support bzip2 and xz whereas Moby's does.